packages/google_maps_flutter/lib/src/marker.dart

```diff
@@ -199,18 +253,51 @@ class MarkerOptions {
       }
     }
 
+    addIfPresent('markerId', markerId.value);
     addIfPresent('alpha', alpha);
     addIfPresent('anchor', _offsetToJson(anchor));
     addIfPresent('consumeTapEvents', consumeTapEvents);
     addIfPresent('draggable', draggable);
     addIfPresent('flat', flat);
     addIfPresent('icon', icon?._toJson());
-    addIfPresent('infoWindowAnchor', _offsetToJson(infoWindowAnchor));
-    addIfPresent('infoWindowText', infoWindowText?._toJson());
+    addIfPresent('infoWindow', infoWindow?._toJson());
     addIfPresent('position', position?._toJson());
     addIfPresent('rotation', rotation);
     addIfPresent('visible', visible);
     addIfPresent('zIndex', zIndex);
     return json;
   }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is Marker &&
+          runtimeType == other.runtimeType &&
+          markerId == other.markerId;
+
+  @override
+  int get hashCode => markerId.hashCode;
+
+  @override
+  String toString() {
+    return 'Marker{markerId: $markerId, alpha: $alpha, anchor: $anchor, '
+        'consumeTapEvents: $consumeTapEvents, draggable: $draggable, flat: $flat, '
+        'icon: $icon, infoWindow: $infoWindow, position: $position, rotation: $rotation, '
+        'visible: $visible, zIndex: $zIndex, onTap: $onTap}';
+  }
+}
+
+Map<MarkerId, Marker> keyByMarkerId(Iterable<Marker> markers) {
+  if (markers == null) {
+    return <MarkerId, Marker>{};
+  }
+  return Map<MarkerId, Marker>.fromEntries(markers.map(
+      (Marker marker) => MapEntry<MarkerId, Marker>(marker.markerId, marker)));
+}
+
+dynamic _serializeMarkerSet(Set<Marker> markers) {
```

Comment:
nit: if this is only used in markers_updates.dart lets keep it in that file.

Also - avoid dynamic when possible.

Reply:
This is also used in `google_map.dart`